File: Leetcode_380.py
import random

class RandomizedSet:

    def __init__(self):
        self.val_to_index = {}
        self.values = []

    def insert(self, val: int) -> bool:
        if val in self.val_to_index:
            return False
        self.val_to_index[val] = len(self.values)
        self.values.append(val)
        return True

    def remove(self, val: int) -> bool:
        if val not in self.val_to_index:
            return False
        index_to_remove = self.val_to_index[val]
        last_val = self.values[-1]
        self.values[index_to_remove] = last_val
        self.val_to_index[last_val] = index_to_remove
        self.values.pop()
        del self.val_to_index[val]
        return True

    def getRandom(self) -> int:
        return random.choice(self.values)

File: Leetcode_380.java
import java.util.*;

class RandomizedSet {
    private Map<Integer, Integer> valToIndex;
    private List<Integer> values;
    private Random rand;

    public RandomizedSet() {
        valToIndex = new HashMap<>();
        values = new ArrayList<>();
        rand = new Random();
    }

    public boolean insert(int val) {
        if (valToIndex.containsKey(val)) {
            return false;
        }
        valToIndex.put(val, values.size());
        values.add(val);
        return true;
    }

    public boolean remove(int val) {
        if (!valToIndex.containsKey(val)) {
            return false;
        }
        int indexToRemove = valToIndex.get(val);
        int lastVal = values.get(values.size() - 1);

        values.set(indexToRemove, lastVal);
        valToIndex.put(lastVal, indexToRemove);

        values.remove(values.size() - 1);
        valToIndex.remove(val);

        return true;
    }

    public int getRandom() {
        int index = rand.nextInt(values.size());
        return values.get(index);
    }
}
